### ISSUE TITLE
Report maxchild value for each service via SNMP

### ISSUE THREAD

**Issue by postilion** [cyrusimap/cyrus-imapd#1827]
Add a new serviceEntry to CYRUS-MASTER.mib with maxchild for each service, so when reading table one may alert on serviceActive as a percentage of serviceMaxchild.

i.e. 
CYRUS-MASTER-MIB::serviceForks.1 = Counter32: 27
CYRUS-MASTER-MIB::serviceActive.1 = Gauge32: 27
CYRUS-MASTER-MIB::serviceName.1 = STRING: imap
CYRUS-MASTER-MIB::serviceId.1 = INTEGER: 1
CYRUS-MASTER-MIB::serviceConnections.1 = Counter32: 31
CYRUS-MASTER-MIB::serviceMaxchild.1 = Counter32: 250

"Service imap 11%; using 27 of 250"

**Comment by elliefm** [comment]
maybe do this at the same time as #1765

**Comment by elliefm** [comment]
@postilion we don't seem to have anything useful in the way of SNMP docs.  All I can find is `doc/legacy/install-snmpmon.html` which I think is both ancient and totally wrong -- it references something called `tugowar`, which looks like it was removed as of _2.1.12(!)_

It sounds like you have it running to some degree or another -- are you able to put together some documentation for it?  Even rough notes would be a help for now -- I'm trying to wrap my head around it but I've never touched it before

**Comment by postilion** [comment]
@elliefm Yeah, one of the things for the To Do list.  I sussed out the proper use of this a couple of months ago, mostly by reading the code, and have been using it in process monitoring since then for a few clients.  One of the things I would like to tackle whilst there, if FastMail even uses it, that is.

**Comment by elliefm** [comment]
We don't at the moment, but would like to once it's in better shape

**Comment by elliefm** [comment]
I think I'll add maxchild to 3.0 too (but the eventual refactor will only be on master)

But in the meantime, I'm putting together some cassandane tests so I can decipher how this thing works, and then verify that I haven't broken it when I start making changes! :P

**Comment by elliefm** [comment]
I've got this working on a side branch, and have the guts of cassandane tests going too (just need some actual tests now):

```
ellie@debian-testing::~$snmpwalk -v1 -cpublic localhost .1.3.6.1.4.1.3.6
CYRUS-MASTER-MIB::cyrusMasterInfoDescr.0 = STRING: Cyrus IMAP server master process
CYRUS-MASTER-MIB::cyrusMasterInfoVers.0 = STRING: 3.1.0-dev-97-g57b6fa4
CYRUS-MASTER-MIB::cyrusMasterInfoUptime.0 = Timeticks: (600) 0:00:06.00
CYRUS-MASTER-MIB::serviceForks.1 = Counter32: 0
CYRUS-MASTER-MIB::serviceActive.1 = Gauge32: 0
CYRUS-MASTER-MIB::serviceName.1 = STRING: imap
CYRUS-MASTER-MIB::serviceId.1 = INTEGER: 1
CYRUS-MASTER-MIB::serviceConnections.1 = Counter32: 0
CYRUS-MASTER-MIB::serviceMaxChildren.1 = INTEGER: 2147483647
```

Interesting to observe that maxchild defaults to INT_MAX if unspecified.  That number surprised me at first!  But I tried it again with imapd's maxchild=6, and got results as expected:

```
ellie@debian-testing::~$snmpwalk -v1 -cpublic localhost .1.3.6.1.4.1.3.6
[...]
CYRUS-MASTER-MIB::serviceMaxChildren.1 = INTEGER: 6
```

**Comment by elliefm** [comment]
FYI, SNMP support is gonna go away when #2100 is merged...

The prometheus implementation on master has included these stats for a while, have you had any time to look at it?

**Comment by dilyanpalauzov** [comment]
Can this be closed?  docsrc/imap/download/release-notes/3.4/x/3.4.0-beta1.rst says “* SNMP support has been removed, as it was broken and unmaintained”

**Comment by elliefm** [comment]
Yeah, I think so.
